core/client/app/controllers/settings/general.js

```diff
@@ -110,6 +124,75 @@ export default Controller.extend(SettingsSaveMixin, {
 
         toggleUploadLogoModal() {
             this.toggleProperty('showUploadLogoModal');
+        },
+
+        validateFacebookUrl(userInput) {
+            let newUrl = userInput.trim();
+            let oldUrl = this.get('model.facebook');
+            let errMessage = '';
+
+            if (!newUrl) {
+                // Clear out the Facebook url
+                this.set('model.facebook', '');
+                return;
+            }
+
+            if (!newUrl.match(/(^https:\/\/www\.facebook\.com\/)(\S+)/g)) {
+                errMessage = 'The URL must be in a format like ' +
+                    'https://www.facebook.com/yourUsername';
+                this.get('model.errors').add('facebook', errMessage);
+                return;
+            }
+
+            // If new url didn't change, exit
+            if (!oldUrl && oldUrl.match(newUrl)) {
```

Comment:
Cursory glance suggests this logic isn't doing what the comment suggests, I think you want to remove the `!` or swap the "or" for an "and"?